providers/tentacle.rb

```diff
@@ -149,60 +148,26 @@
   api_key = new_resource.api_key
   roles = new_resource.roles
   environment = new_resource.environment
+  config_path = new_resource.config_path
 
   verify_server(server)
   verify_api_key(api_key)
   verify_roles(roles)
   verify_environment(environment)
 
-  # Itterate over every role and make a big long string like:
-  # --role "web" --role "database" --role "app-server"
-  role_expression = ''
-  roles.each do |a_role|
-    role_expression << "--role \"#{a_role}\" "
+  register_instance = powershell_script "register-#{instance}-tentacle" do
+    action :run
+    cwd tentacle_install_location
+    code <<-EOH
+      .\\Tentacle.exe register-with --instance "#{instance}" --server "#{server}" --name "#{node.name}" --apiKey "#{api_key}" #{register_comm_config(polling, port)} --environment "#{environment}" #{option_list('role', roles)} --console
+      #{catch_powershell_error('Registering Tentacle')}
+    EOH
+    # This is sort of a hack, you need to specify the config_path on register if it is not default
+    # The other option is to read the registry key but the helpers are not available in 12.4.1
+    not_if { tentacle_exists?(server, api_key, tentacle_thumbprint(config_path)) }
```

Comment:
The tentacle path is in the registry key so ideally looking there would be prefered since its created when the service is created.  I think this will require a new minimum version of chef-client.  Once v1.0.0 is closer I am going to limit this to > 12.5.1 so we get more features without all the legacy items.